sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -555,6 +557,61 @@ class HiveDDLSuite
     }
   }
 
+  test("Alter Table Set Location") {
+    withTable("tab1", "tab2") {
+      val catalog = spark.sessionState.catalog
+      sql("CREATE TABLE tab1 using parquet AS SELECT 1 as a")
+      sql("CREATE TABLE tab2 using parquet AS SELECT 2 as a")
+      checkAnswer(spark.table("tab1"), Seq(Row(1)))
+      checkAnswer(spark.table("tab2"), Seq(Row(2)))
+      val metadataTab1 = catalog.getTableMetadata(TableIdentifier("tab1"))
+      val locTab1 = metadataTab1.storage.locationUri
+      val metadataTab2 = catalog.getTableMetadata(TableIdentifier("tab2"))
+      val locTab2 = metadataTab2.storage.locationUri
+      assert(locTab1.isDefined)
+      assert(locTab2.isDefined)
+      try {
+        sql(s"ALTER TABLE tab2 SET LOCATION '${locTab1.get}'")
+        checkAnswer(spark.table("tab2"), Seq(Row(1)))
+      } finally {
+        val root = new Path(locTab2.get)
+        val fs = root.getFileSystem(spark.sparkContext.hadoopConfiguration)
+        fs.delete(root, true)
```

Comment:
@mallman After altering the table location, `locTab1` is not the location of the `tab2`. Thus, when we dropping the table `tab2`, the location is not automaticaly deleted. Here, we need to manually delete it; otherwise, the following test case is unable to create a managed table using the name `tab1`.